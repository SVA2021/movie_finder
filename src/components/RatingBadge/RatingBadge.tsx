import {Typography} from "@mui/material";
import {FC} from "react";
import {getColorFromRate, getRateFromString} from "../../utils";

interface RatingBadgeProps {
  rating: number | string | null | undefined
  position: 'left' | 'right'
}

export const RatingBadge: FC<RatingBadgeProps> = ({rating, position}) => {

  const normalizeRating = getRateFromString(rating);
  const bgColor = getColorFromRate(normalizeRating);

  return (
    <Typography
      data-testid={'rating-badge'}
      variant="h3"
      component="p"
      sx={{
        position: 'absolute',
        top: 0,
        left: position === 'left' ? 0 : null,
        right: position === 'right' ? 0 : null,
        padding: '0px 12px',
        backgroundColor: bgColor,
        color: 'common.white',
      }}>
      {normalizeRating?.toFixed(1)}
    </Typography>
  )
};