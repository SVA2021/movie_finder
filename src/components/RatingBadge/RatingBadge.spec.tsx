import {render, screen} from "@testing-library/react";
import {RatingBadge} from "./RatingBadge";

describe('render RatingBadge', () => {

  const setUp = () => render(<RatingBadge rating={7.0} position={"left"} />);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('rating-badge')).toMatchSnapshot();
  });

  it('check text', () => {
    setUp();
    expect(screen.getByText(/7/i)).toBeInTheDocument();
  });

})