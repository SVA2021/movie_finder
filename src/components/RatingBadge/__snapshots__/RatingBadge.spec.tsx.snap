// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render RatingBadge snapshot test 1`] = `
<p
  class="MuiTypography-root MuiTypography-h3 css-1uovp3e-MuiTypography-root"
  data-testid="rating-badge"
>
  7.0
</p>
`;
